### DISCUSSION TITLE
Horizontal scrollbars on code view are off the screen

### DISCUSSION THREAD

**Discussion by simonsabin** [community/community#5441] [Other Feature Feedback, Questions, & Ideas]
Love github but something that is always frustrating is that when you have code that is wide and longer than a page the horizontal scroll bars are off the screen. To use them one has to use the main page vertical scroll bar to find the horizontal scroll bar, scroll and then scroll back using the page vertical scroll. 

This doesn't work on so many points
* you can't see the code that you are trying to scroll to when scrolling
* once you've scrolled you've got to do everything in reverse to see the main body of code
* if you have long code you can find your having to use the scrollbars in the code window and on the page

I'd like one of two things, 
* the code window take up a maximum size of page thus the scrollbars appear on the page
* the horizontal scroll bar floats in someway thus having it always in view
* an option to turn on wrapping lines (this is what seems to be done by default on the PR code diff view)

**Comment by simonsabin** [comment]
Another option is to enable the code view to fill the width of the screen rather than be constrained to only a sub section in the middle of the page 
![image](https://user-images.githubusercontent.com/1209963/131980059-25afe023-b87c-4140-b0bf-4a0680c62b3e.png)

**Reply by kedicesur** [reply]
I use the [Wide GitHub](https://chrome.google.com/webstore/detail/wide-github/kaalofacklcidaampbokdplbklpeldpj) extension of Chrome to achieve this functionality. It proves to be an essential tool for my wide indenting style.

**Comment by emmaviolet** [comment]
Just a note to say that we're looking into this - will get back

**Reply by simonsabin** [reply]
Cheers

**Comment by shmeadyy** [comment]
Hi @simonsabin! Thanks for letting us know about this issue. We did some initial development to try to solve for it and unfortunately ran into some caching issues that need to be taken care of first before we can continue development on this. So for now we're going to have to wait to fix it with an undetermined fix date, but we'll be sure to keep you posted should that change. Thanks for your understanding!

**Reply by simonsabin** [reply]
That’s a shame. Look forward to a fix in the future

**Reply by shmeadyy** [reply]
I know, I was disappointed too. Will keep you posted for a future fix!

**Comment by kedicesur** [comment]
@shmeadyy Despite it's not any popular (or even known), I use [wide indenting in my codes](https://github.com/kedicesur/AQ/blob/master/lib/async-queue.js) which for me, makes them so very much easy to follow. I have ultra wide (21:9) monitors and I currently use the [Wide GitHub](https://chrome.google.com/webstore/detail/wide-github/kaalofacklcidaampbokdplbklpeldpj) Chrome extension. So I am OK but had this been a native option (like dark mode) it would be very easy to tell people to switch on the wide mode for proper display when I share a link to my repo. However still it would be nice if you could implement horizontal scroll of the code by clicked (pressed) scroll wheel for instance.

**Comment by jfullstackdev** [comment]
there was a lengthy discussion with this in the past, it's just being mobile-first

actually GitHub UI improved a lot when it comes to changing the size of the window opened, it's being responsive, well it's not only that wide space, even Twitter does the same thing

![image](https://user-images.githubusercontent.com/47092464/200177313-19f2d194-e54a-4b3a-9b67-3a911511f20e.png)

and the message tab is actually dangling there at the bottom as if that does not belong to the group

for ultrawide monitors, of course even videos in a 49" _computer monitor_ will not be fully consuming the entire screen, we all know that

it's either setting the monitor to be divided or manually resizing to maximize the screen accommodating many windows

**Comment by emmaviolet** [comment]
Just adding a note here to capture some discussions on Slack.

Unfortunately we tried to improve this experience back in February, but found that it was much more complicated than we expected due to various accessibility concerns, mostly with wrapping code in code blocks, which was the option above that we explored the most fully. I hope to be able to revisit this at some point, but regret at this point we have no further updates.
